lib/rex/ui/text/dispatcher_shell.rb

```diff
@@ -588,7 +588,15 @@ def run_command(dispatcher, method, arguments)
   # If the command is unknown...
   #
   def unknown_command(method, line)
-    print_error("Unknown command: #{method}")
+    # Map each dispatchers commands to valid_commands
+    valid_commands = dispatcher_stack.flat_map { |dispatcher| dispatcher.commands.keys }
+
+    message = "Unknown command: #{method}."
+    suggestion = DidYouMean::SpellChecker.new(dictionary: valid_commands).correct(method).first
+    message << " Did you mean #{suggestion}?" if suggestion
```

Comment:
what do you think about highlighting the suggested command to help it stick out similar to how `help` is highlighted below?

Reply:
I think that's a good idea. Went ahead and added that. Thanks!